### PR TITLE
feat: Use Isolate.register_op in deno_typescript

### PR DESCRIPTION
**Pull request by bartlomieju**
Built upon #3039 

Added op registration to `deno_typescript`.

Also removed `set_dispatch` and effectively "dual-dispatch" mechanism, so fingers crossed for bump in benchmarks 🤞

### CODE REVIEW

**Comment by bartlomieju** [comment]
Benchmarks

<details><summary>master</summary>
<p>

```
  "req_per_sec": {
    "deno_tcp": 50893, 
    "deno_core_single": 52577, 
    "node_proxy_tcp": 21849, 
    "deno_proxy_tcp": 17559, 
    "deno_core_multi": 61955, 
    "deno_tcp_current_thread": 48172, 
    "hyper": 64200, 
    "node_tcp": 62553, 
    "deno_http": 17850, 
    "node_http": 25033, 
    "deno_proxy": 946, 
    "node_proxy": 2797
  }, 
  "max_latency": {
    "deno_tcp": 10.32, 
    "deno_core_single": 0.421, 
    "node_proxy_tcp": 1.05, 
    "deno_proxy_tcp": 6.58, 
    "deno_core_multi": 1.01, 
    "deno_tcp_current_thread": 10.55, 
    "hyper": 0.87, 
    "node_tcp": 0.604, 
    "deno_http": 3.04, 
    "node_http": 1.9, 
    "deno_proxy": 18.6, 
    "node_proxy": 10.61
  }, 
```

</p>
</details>

<details><summary>
this PR
</summary>
<p>

```
  "req_per_sec": {
    "deno_tcp": 51645, 
    "deno_core_single": 54635, 
    "node_proxy_tcp": 22353, 
    "deno_proxy_tcp": 18260, 
    "deno_core_multi": 53625, 
    "deno_tcp_current_thread": 50520, 
    "hyper": 64628, 
    "node_tcp": 66266, 
    "deno_http": 18312, 
    "node_http": 25909, 
    "deno_proxy": 983, 
    "node_proxy": 2858
  }, 
  "max_latency": {
    "deno_tcp": 9.16, 
    "deno_core_single": 0.402, 
    "node_proxy_tcp": 1.0, 
    "deno_proxy_tcp": 5.97, 
    "deno_core_multi": 1.0, 
    "deno_tcp_current_thread": 9.45, 
    "hyper": 1.48, 
    "node_tcp": 0.566, 
    "deno_http": 2.82, 
    "node_http": 2.06, 
    "deno_proxy": 18.01, 
    "node_proxy": 9.88
  },
```

</p>
</details>

File: js/os.ts [hunk 1]
```diff
@@ -81,7 +81,14 @@ interface Start {
...
   core.setAsyncHandler(dispatch.asyncMsgFromRust);
-
+  const ops = core.ops();
+  // TODO(bartlomieju): this is a prototype, we should come up with
+  // something a bit more sophisticated
+  for (const [name, opId] of Object.entries(ops)) {
+    const opName = `OP_${name.toUpperCase()}`;
+    // Assign op ids to actual variables
+    dispatch[opName] = opId;
+  }
```
Comment (ry) [inline js/os.ts]: I agree that this isn't the most ideal solution, but I think it will work for now.

**Comment by ry** [comment]
I think this looks pretty good. Certainly we have more work to do before stabilizing the op interface, but this is an improvement.

@bartlomieju could you please draft a commit message?

**Comment by bartlomieju** [comment]
@ry this PR includes changes from #3039 and is not rebased. Maybe we could land that PR first?

**Comment by bartlomieju** [comment]
@ry rebased on top of master, feel free to push, I'm out till tomorrow

**Comment by ry** [comment]
Just waiting to sample more benchmarks to make sure #3039 didn't introduce regressions. Otherwise LGTM.

**Pull request merged**